File: reactui/src/common/models/merchant.ts
import { ISoldier } from "./soldier";
import { IPort } from "./port";
import { ICargo } from "./cargo";

export interface IMerchant {
    merchantId: string;
    cargo: ICargo | null;
    port: IPort;
    soldiers: ISoldier[];
    status: string;
    name: string;
    skill: number;
    loyalty: number;
    resources: number;
    age: number;
}

File: Domain/Entities/Persons/Merchant.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities.Persons
{
    public class Merchant
    {
        public Merchant()
        {
            Soldiers = new HashSet<Soldier>();
        }

        public Guid MerchantId { get; set; }
#nullable enable
        public virtual Cargo? Cargo { get; set; }
#nullable disable
        public virtual Port Port { get; set; }
        public virtual ICollection<Soldier> Soldiers { get; set; }
        public string Status { get; set; }
        public string Name { get; set; } = "";
        public int Skill { get; set; }
        public int Resources { get; set; }
        public int Loyalty { get; set; }
        public int Age { get; set; }
    }
}